templates/cli/lib/commands/init.js.twig

```diff
@@ -71,20 +79,48 @@ const initFunction = async () => {
     parseOutput: false
   })
 
-  let command = `
-    mkdir -m 777 -p 'functions/${answers.name}' && \
-    cd 'functions/${answers.name}' && \
-    git init && \
-    git remote add -f origin https://github.com/{{ sdk.gitUserName }}/functions-starter && \
-    git config core.sparseCheckout true && \
-    echo '${answers.runtime.id}' >> .git/info/sparse-checkout && \ 
-    git pull origin main && \
-    rm -rf .git && \
-    mv ${answers.runtime.id}/* . && \
-    rm -rf ${answers.runtime.id}`;
+  fs.mkdirSync(functionDir, "777");
+
+  const gitInitCommands =
+    `git init && ` +
+    `git remote add -f origin https://github.com/{{ sdk.gitUserName }}/functions-starter && ` +
+    `git config core.sparseCheckout true`;
```

Comment:
Can the various git commands be reduced to something like:

```javascript
const gitClone = "git clone --depth 1 --sparse https://github.com/{{ sdk.gitUserName }}/functions-starter"; // depth prevents fetching older commits reducing the amount fetched
childProcess.execSync(gitClone, { stdio: 'pipe' });
const gitCheckout = `git sparse-checkout add `${answers.runtime.id}`;
childProcess.execSync(gitCheckout, { cwd: "functions-starter", stdio: 'pipe' }); // use cwd option to avoid cd
```

Reply:
@PineappleIOnic, thoughts on this?

Reply:
This is an excellent idea! Thank you for bringing these git commands to my attention.

My apologies for not responding sooner, drafting the changes now...